lib/redis_failover.rb

```
@@ -1,4 +1,10 @@
 require 'zk'
+
+#NOTE: We've found that using the 'recommended' zk fork-hook would trigger
+#kernel mutex deadlocks in forking env (unicorn & resque) [ruby 1.9]
+#https://github.com/zk-ruby/zk/wiki/Forking & https://github.com/zk-ruby/zk/blob/master/RELEASES.markdown#v150
+#ZK.install_fork_hook
```

this is insanely dangerous and will break _worse_ with this disabled. Safe operation _requires_ the use of the fork hooks to ensure that the mutexes are owned by the thread calling fork() _and_ the child is able to clean up the connection properly.

This is likely just masking the error or causing a different kind of broken behavior. The latest release of zk-1.9.3 & zookeeper-1.4.8 have fixes for the event delivery system that we believe caused the deadlocks (@eric has been running them with success in production for several weeks)

You _must_ install the fork hooks for correct operation in forking environments.